mesmerize_viz/_common.py

```diff
@@ -0,0 +1,185 @@
+from typing import *
+from functools import partial
+
+import numpy as np
+import pandas as pd
+from fastplotlib import ImageWidget
+from fastplotlib.utils import quick_min_max
+from mesmerize_core import MCorrExtensions
+
+from ._utils import ZeroArray
+
+
+class ImageWidgetWrapper:
+    """Wraps Image Widget in a way that allows updating the data"""
+    def __init__(
+            self,
+            data: List[str],
+            data_mapping: dict,
+            standard_mappings: List[str],
+            reset_timepoint_on_change: bool = False,
+            input_movie_kwargs: dict = None,
+            image_widget_kwargs: dict = None,
+    ):
+        """
+        Visualize motion correction output.
+
+        Parameters
+        ----------
+        data: list of str, default ["input", "mcorr"]
+            list of data to plot, can also be a list of lists.
+
+        reset_timepoint_on_change: bool, default False
+            reset the timepoint in the ImageWidget when changing items/rows
+
+        input_movie_kwargs: dict, optional
+            kwargs passed to get_input_movie()
+
+        image_widget_kwargs: dict, optional
+            kwargs passed to ImageWidget
+
+        Returns
+        -------
+        ImageWidget
+            fastplotlib.ImageWidget visualization
+        """
+
+        if input_movie_kwargs is None:
+            input_movie_kwargs = dict()
+
+        if image_widget_kwargs is None:
+            image_widget_kwargs = dict()
+
+        # the ones which are [t, x, y] images that ImageWidget can manage by itself
+        self.standard_mappings = standard_mappings
```

Comment:
I just made the name "standard_mappings" for things that are [t, x, y] arrays, like input or mcorr movies since they can be managed together by image widget (since the dims match), but mean, corr, etc and in the future temporal  cannot be managed by ImageWidget. Maybe I should call it "image_widget_managed_data"

Reply:
data mapping is a dict where the data labels, such as "input", "mcorr", "mean" map to the extension functions that return the corresponding data array.